woocommerce/woocommerce-android
# Store Creation: Better country selector part I: UI and navigation

Part of: https://github.com/woocommerce/woocommerce-android/issues/9606


### Description


This PR implements a design update and navigation for the country selector steps during store creation. It also deletes unneeded parts in the code that's part of the old flow.

**Please note that the PR is only concerned with UI changes.** The PR is already large as-is and I don't want to make it even bigger. I'm setting this as do-not-merge for now, waiting for the next PR #9607 to be merged first.

<details><summary>Sorry that this PR is quite large! 🧵 </summary>
A lot of this PR is just moving and lightly editing existing code for the country list picker from the old flow, to the new flow, and then deleting the unused code from the old flow.

I added some notes to the beginning of some files, to hopefully help explain what's in them and make reviewing easier.
</details> 

| comparison | screenshot |
|-|-|
| old flow | <img src="https://github.com/woocommerce/woocommerce-android/assets/266376/519c7360-fa55-4a15-b815-bb73c98903e9" width="42%" />  |
| new flow | <img src="https://github.com/woocommerce/woocommerce-android/assets/266376/fa521cd7-2b98-480b-922b-1ee9ef7de537" /> |

### Testing instructions

1. Create a new site, skip the profiler questions,
2. Ensure the first screen for the country picker is shown, similar to screenshot above.
3. Tap the shown country, ensure that the second screen for the country picker (with the country list) is shown,
4. Play around and test the scrolling in the list of countries. Also, in potrtrait mode, the "Current location" area will stick, while in landscape mode, the "Current location" area will scroll. This is to match the same behavior in the old flow.
5. Tap back button and ensure that you're returned to the first screen.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/266376/edcdb3da-d6dd-43cd-bf78-31ab150497d1
